packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -73,6 +78,64 @@ const {
 
 const language = 'TypeScript';
 
+function translateObjectTypeAnnotation(
+  hasteModuleName: string,
+  /**
+   * TODO(T108222691): Use flow-types for @babel/parser
+   */
+  nullable: boolean,
+  objectMembers: $ReadOnlyArray<$FlowFixMe>,
+  typeAliasResolutionStatus: TypeAliasResolutionStatus,
+  baseTypes: $ReadOnlyArray<string>,
+  types: TypeDeclarationMap,
+  aliasMap: {...NativeModuleAliasMap},
+  tryParse: ParserErrorCapturer,
+  cxxOnly: boolean,
+  parser: Parser,
+): Nullable<NativeModuleTypeAnnotation> {
+  // $FlowFixMe[missing-type-arg]
+  const properties = objectMembers
+    .map<?NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>>(property => {
+      return tryParse(() => {
+        return parseObjectProperty(
+          property,
+          hasteModuleName,
+          types,
+          aliasMap,
+          tryParse,
+          cxxOnly,
+          nullable,
+          translateTypeAnnotation,
+          parser,
+        );
+      });
+    })
+    .filter(Boolean);
+
+  if (baseTypes.length === 0) {
```

Comment:
can we simplify and avoid some code duplication with:

```ts
let objectTypeAnnotation;
if (baseTypes.length === 0) {
  objectTypeAnnotation = {
        type: 'ObjectTypeAnnotation',
        properties,
      }
} else {
  objectTypeAnnotation = {
        type: 'ObjectTypeAnnotation',
        properties,
        baseTypes,
      }
}

return typeAliasResolution(
      typeAliasResolutionStatus,
      objectTypeAnnotation,
      aliasMap,
      nullable,
);
```

also, if we are using this `if length === 0 then simpleObjectTypeAnnotation else objectTypeAnnotationWithBaseType` logic in more places, perhaps we can add a factory method for it. Something like

```ts
function createObjectTypeAnnotation(properties, baseTypes) {
if (baseTypes.length === 0) {
  return {
        type: 'ObjectTypeAnnotation',
        properties,
      };
} else {
  return {
        type: 'ObjectTypeAnnotation',
        properties,
        baseTypes,
      };
}
```

and then use it like:

```ts
return typeAliasResolution(
      typeAliasResolutionStatus,
      createObjectTypeAnnotation(properties, baseTypes),
      aliasMap,
      nullable,
    );
```

What do you think?

Reply:
Fixed. I don't think we are going to repeat this logic else where so I didn't extract the function.